src/Arcus.Observability.Telemetry.Core/AssemblyAppVersion.cs

```diff
@@ -1,13 +1,15 @@
 ﻿using System;
 using System.Reflection;
-using GuardNet;
 
 namespace Arcus.Observability.Telemetry.Core
 {
     /// <summary>
     /// Represents an <see cref="IAppVersion"/> implementation that uses the assembly version as application version.
     /// </summary>
     /// <seealso cref="IAppVersion"/>
+#pragma warning disable S1133
+    [Obsolete("Will be removed in v4.0 as application versioning enrichment is too project-specific")]
+#pragma warning restore S1133
```

Comment:
I don't think we should do that.  IMHO, this is really valuable.

Reply:
Maybe in the older Serilog setup, but with OpenTelemetry these 'resource information' is set differently and on different places, as well. Think their way of working is rather clean and interfaces like these would maybe overcomplicate it.